core/checker.py

```diff
@@ -105,8 +117,9 @@ def wrapped_callback(event_type, data):
             )
             try:
                 step_result = step_instance.check_with_messages()
-            except SyntaxError:
-                pass
+            except SyntaxError as syntax_error:
+                print(f"SyntaxError: {syntax_error}")
+                raise syntax_error
```

Comment:
I think both of these changes with `SyntaxError` should be reverted. This particular line (reraising the error instead of ignoring it) is a major change in behaviour that I don't think is desirable. I also don't think it's a good idea to print in either case, what's the reasoning?